### PR TITLE
Bulk upload designations upload fixes

### PR DESCRIPTION
**Pull request by Sohith-code**

### CODE REVIEW

File: project/ws/app/src/lib/routes/home/components/request-list/create-request-form/create-request-form.component.ts [hunk 1]
```diff
@@ -386,7 +386,7 @@ export class CreateRequestFormComponent implements OnInit {
   compAreaSelected(option: any) {
     this.resetCompSubfields()
     this.allCompetencies.forEach((val: any) => {
-      if (option.name === val.name) {
+      if (option.name === val.name && val.children.length) {
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/request-list/create-request-form/create-request-form.component.ts]: please check for val.children before val.children.length

File: project/ws/app/src/lib/routes/home/components/request-list/create-request-form/create-request-form.component.ts [hunk 2]
```diff
@@ -398,7 +398,7 @@ export class CreateRequestFormComponent implements OnInit {
   compThemeSelected(option: any) {
     this.enableCompetencyAdd = false
     this.allCompetencyTheme.forEach((val: any) => {
-      if (option.name === val.name) {
+      if (option.name === val.name && val.children.length) {
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/components/request-list/create-request-form/create-request-form.component.ts]: same as above

File: project/ws/app/src/lib/routes/home/components/request-list/create-request-form/create-request-form.component.ts (hunk 1 above)
Comment (Sohith-code) [reply project/ws/app/src/lib/routes/home/components/request-list/create-request-form/create-request-form.component.ts]: Added the check

File: project/ws/app/src/lib/routes/home/components/request-list/create-request-form/create-request-form.component.ts (hunk 2 above)
Comment (Sohith-code) [reply project/ws/app/src/lib/routes/home/components/request-list/create-request-form/create-request-form.component.ts]: Added the check

**Pull request merged**